extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -170,92 +178,98 @@ public ScaleStatus GetScaleStatus(ScaleStatusContext<RabbitMQTriggerMetrics> con
         return this.GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
     }
 
-    private static bool IsTrueForLast(IList<RabbitMQTriggerMetrics> samples, int count, Func<RabbitMQTriggerMetrics, RabbitMQTriggerMetrics, bool> predicate)
+    /// <summary>
+    /// Returns scale recommendation i.e. whether to scale in or out the host application. The recommendation is
+    /// made based on both the latest metrics and the trend of increase or decrease in the the number of messages in
+    /// Ready state in the queue. In all of the calculations, it is attempted to keep the number of workers minimum
+    /// while also ensuring that the message count per worker stays under the maximum limit.
+    /// </summary>
+    /// <param name="workerCount">The current worker count for the host application.</param>
+    /// <param name="metrics">The collection of metrics samples to make the scale decision.</param>
+    private ScaleStatus GetScaleStatusCore(int workerCount, RabbitMQTriggerMetrics[] metrics)
     {
```

Comment:
I have replaced the primitive scaling logic with one that I wrote for SQL extension. There were couple of issues with the existing implementation, primarily it used to scale out too aggressively.

Reply:
This is not complete implementation of runtime-driven scaling, right? Just an improvement on the current implementation, and we will need another PR for runtime-driven scaling?

Reply:
This is actually runtime driven scaling logic. We do not have RabbitMQ scaling logic encoded  inside Scale controller which prevents scaling support for consumption tier, and on dedicated tiers unless customer explicitly enable runtime driven scaling. By the way, I still need to test the update scaling implementation on real service. I will complete the PR after ensuring that scaling works as expected.

Reply:
It's really great if the scaling works without hiccups. We should get the extensions bundles updated after that.

Reply:
There was a bug but otherwise I could ensure that scaling works fine i.e., the number of instances increases on high load, it stays consistent with expected worker-count for continuous load and goes back to one (Elastic Premium behavior) when the queue becomes idle. Please review the new commit.